src/consensus/sync_block.cpp

```diff
@@ -11,24 +11,22 @@ namespace taraxa {
 
 using namespace std;
 
-SyncBlock::SyncBlock(PbftBlock const& pbft_blk, std::vector<std::shared_ptr<Vote>> const& cert_votes)
-    : pbft_blk(new PbftBlock(pbft_blk)), cert_votes(cert_votes) {}
+SyncBlock::SyncBlock(std::shared_ptr<PbftBlock> pbft_blk, std::vector<std::shared_ptr<Vote>> cert_votes)
+    : pbft_blk(std::move(pbft_blk)), cert_votes(std::move(cert_votes)) {}
```

Comment:
When I look at this, it seems that passing const ref to vector and then making a copy `cert_votes(cert_votes)` is better/cheaper than passing it as copy and then moving it ?

It is basically 1 reference + 1 copy ctor VS 1 copt ctr + 1 move ctor

Reply:
it was making scene before but lot of changed since this PR